src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -142,6 +142,8 @@ public async Task<AuthenticationResult> ExecuteAsync(
                 commonParameters,
                 requestContext,
                 _confidentialClientApplication.UserTokenCacheInternal).ConfigureAwait(false);
+            
+            requestParams.SendX5C = usernamePasswordParameters.SendX5C ?? false;
```

Comment:
I was also wondering whether we could/should put this setting into a central place when `WithCertificate(...)` is called, but then I realized that it has been discussed and [the conclusion](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4917/files#r1747523539) makes sense.

Ship it!